src/app/_common/data/adData.js

```diff
@@ -0,0 +1,243 @@
+        applicationEmail: getEmail(properties.applicationemail),
+        applicationUrl: getUrl(properties.applicationurl),
+        sourceUrl: getUrl(properties.sourceurl),
+        hasSuperraskSoknad: getString(properties, "hasInterestform"),
+        jobPostingFormat: getJobPostingFormat(properties.adText),
+
+        // employment details
+        engagementType: getString(properties.engagementType),
+        extent: getString(properties.extent),
+        jobArrangement: getString(properties.jobarrangement),
+        jobPercentage: getJobPercentage(properties.jobpercentage),
+        jobTitle: getString(properties.jobtitle),
+        positionCount: getString(properties.positioncount),
+        remote: getString(properties.remote),
+        sector: getString(properties.sector),
+        startTime: getString(properties.starttime),
+        workday: getWorktime(properties.workday),
+        workhours: getWorktime(properties.workhours),
+        workLanguages: getArray(properties.workLanguage),
+        locationList: getLocationListData(data.locationList),
+        location: getString(properties.location),
+
+        // Employer
+        employer: getEmployerData(data),
+        contactList: getContactList(data.contactList),
+    };
+}
+
+function getAdText(adText) {
+    let processedAdText = adText;
+    if (containsEmail(adText)) {
+        try {
+            const extractedEmails = [...extractEmail(adText)];
+            let preprocessedAd = adText.replace(/&#64;/g, "@");
+            extractedEmails.forEach((it) => {
+                if (isValidEmail(it) && !mailtoInString(preprocessedAd, it)) {
+                    preprocessedAd = preprocessedAd.replace(it, `<a rel="nofollow" href="mailto:${it}">${it}</a>`);
+                }
+            });
+            processedAdText = preprocessedAd;
+        } catch (err) {
+            processedAdText = adText;
+        }
+    }
+    // TODO: double check that sanitizing doesn't remove <section> (see amplitude.js for usage)
+    return DOMPurify.sanitize(processedAdText);
+}
+
+function getEmail(email) {
+    return isValidEmail(email) ? email : undefined;
+}
+
+function getJobPercentage(value) {
+    const jobPercentage = getString(value);
+    if (!jobPercentage) {
+        return undefined;
+    }
+
+    return jobPercentage + jobPercentage.endsWith("%") ? "" : "  %";
+}
+
+function getWorktime(value) {
+    const worktime = getString(value);
+    if (!worktime) {
+        return undefined;
+    }
+
+    // We need this check in case of old workhour/-day property values, formatted like 'Opt1 Opt2'
+    const items = [];
+    try {
+        const jsonArray = JSON.parse(worktime);
+
+        for (let i = 0; i < jsonArray.length; i += 1) {
+            items.push(jsonArray[i]);
+        }
+    } catch (e) {
+        items.push(worktime);
+    }
+    return items.join(", ");
+}
+
+function getContactList(contactList) {
+    if (!contactList) {
```

Comment:
bruke `const contactList = getArray(value)` her for å også sjekket at det faktisk er en array, for å unngå at `map()` under kan feile hvis ikke array